libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -286,79 +287,113 @@ export class HttpRequest<O extends object = {}> extends Dialog<O> implements Htt
-                    result.content = JSON.parse(content);
-                } catch {
-                    result.content = content;
+            const result = new Result(response.headers);
+            result.statusCode = response.status;
+            result.reasonPhrase = response.statusText;
+
+            switch (this.responseType.getValue(dc.state)) {
+                case ResponsesTypes.Activity:
+                    result.content = await response.json();
+                    dc.context.sendActivity(result.content as Activity);
+                    break;
+                case ResponsesTypes.Activities:
+                    result.content = await response.json();
+                    dc.context.sendActivities(result.content as Activity[]);
+                    break;
+                case ResponsesTypes.Json: {
+                    const content = await response.text();
+                    try {
+                        result.content = JSON.parse(content);
+                    } catch {
+                        result.content = content;
+                    }
+                    break;
                 }
-                break;
+                case ResponsesTypes.Binary: {
+                    const buffer = await response.arrayBuffer();
+                    result.content = new Uint8Array(buffer);
+                    break;
+                }
+                case ResponsesTypes.None:
+                default:
+                    break;
             }
-            case ResponsesTypes.Binary: {
-                const buffer = await response.arrayBuffer();
-                result.content = new Uint8Array(buffer);
-                break;
+
+            traceInfo.response = result;
+
+            await this.EndDialogWithResultAsync(dc, result, traceInfo);
+        } catch (err) {
+            const result = new Result();
+            result.content = err.message;
+            traceInfo.response = result;
+
+            if (err instanceof FetchError) {
+                result.statusCode = StatusCodes.NOT_FOUND;
+            } else {
+                throw err;
             }
-            case ResponsesTypes.None:
-            default:
-                break;
+
+            await this.EndDialogWithResultAsync(dc, result, traceInfo);
         }
+    }
 
-        traceInfo.response = result;
+    /**
+     * Writes Trace Activity for the http request and response values and returns the actionResult as the result of this operation.
+     *
+     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param result Optional. Value returned from the dialog that was called. The type
+     * of the value returned is dependent on the child dialog.
+     * @param traceInfo Optional. The text to be spoken by your bot on a speech-enabled channel.
+     * @returns A `Promise` representing the asynchronous operation.
+     */
+    private async EndDialogWithResultAsync(
```

Comment:
The names of the methods in JS differ a little from .NET.
```suggestion
    private async endDialogWithResult(
```

Reply:
Done. Thanks!